src/proxy/server.rs

```diff
@@ -681,24 +483,25 @@ mod tests {
                 let proxy_metrics = proxy_metrics.clone();
                 let session_metrics = SessionMetrics::new(&metrics.registry).unwrap();
 
-                worker_configs.push(DownstreamReceiveWorkerConfig {
-                    worker_id,
-                    packet_rx,
-                    shutdown_rx: shutdown_rx.clone(),
-                    receive_config: ProcessDownstreamReceiveConfig {
-                        proxy_metrics: proxy_metrics.clone(),
-                        session_metrics,
-                        cluster_manager: cluster_manager.clone(),
-                        filter_manager: filter_manager.clone(),
-                        session_manager: session_manager.clone(),
-                        session_ttl: Duration::from_secs(10),
-                        send_packets: send_packets.clone(),
-                    },
-                })
+                tokio::spawn(
```

Comment:
More of an organisational thing, but since this is testing the `DownstreamReceiveWorker` and the `DownstreamReceiveProcessor` - I figure this test should get moved to `dowmstream.rs` to make it more discoverable.

Also probably renamed, since we no longer have a function called `spawn_downstream_receive_workers` 🙂